src/test/java/se/liu/ida/hefquin/queryplan/executable/impl/ops/ExecOpSymmetricHashJoinTest.java

```diff
@@ -0,0 +1,207 @@
+package se.liu.ida.hefquin.queryplan.executable.impl.ops;
+
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.junit.Test;
+import se.liu.ida.hefquin.data.SolutionMapping;
+import se.liu.ida.hefquin.data.jenaimpl.JenaBasedSolutionMapping;
+import se.liu.ida.hefquin.data.jenaimpl.JenaBasedSolutionMappingUtils;
+import se.liu.ida.hefquin.queryplan.executable.impl.GenericIntermediateResultBlockImpl;
+import se.liu.ida.hefquin.queryplan.executable.impl.MaterializingIntermediateResultElementSink;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.*;
+
+public class ExecOpSymmetricHashJoinTest extends ExecOpTestBase
+{
+	@Test
+	public void JoinWithOneJoinVariable(){
```

Comment:
Also in this file, please adhere to the usual naming conventions of method names.